### DISCUSSION TITLE
[Design] Pulsar All Releases Page

### DISCUSSION THREAD

**Discussion by Anonymitaet** [apache/pulsar#17310] [Ideas]
Hi team,

1. How about providing more useful content to the [Pulsar Release Notes > All Releases](https://pulsar.apache.org/release-notes/) page as [below](https://docs.google.com/spreadsheets/d/1XHmzno-1kbHvPp7eogri8XAiShH-EgpEG40svte-qxM/edit#gid=1677243891)?

<img width="1029" alt="image" src="https://user-images.githubusercontent.com/50226895/187336497-95a68232-36e5-4fc7-a0c0-7b3788ba78a6.png">

2. This table should be updated automatically. 

Info | Info can be pulled from
---|---
Release note URL | https://pulsar.apache.org/release-notes/
Release blogs name and URL | https://pulsar.apache.org/blog
Documentation URL | [Versions page](https://pulsar.apache.org/versions), so this page can be removed since the info is consolidated into the `All Releases` page.
Release date </br> Release manager| https://github.com/apache/pulsar/releases.

☘️☘️☘️

Hi @tisonkun @urfreespace @SignorMercurio thoughts?

**Comment by SignorMercurio** [comment]
Looks great, though not clear on how to update the table from releases.

**Reply by urfreespace** [reply]
we can get the release information by `gh` cli, but I don't think you can get the `release date` and `release manager` info from that. Maybe you need to add them manual, pls reference here about `gh` using https://github.com/apache/pulsar-site/blob/760f7b47f437267a3c9d71d14c73e99ba32bd1ec/site2/website-next/scripts/release-notes.sh#L42. @SignorMercurio @Anonymitaet

**Reply by tisonkun** [reply]
I think the release manager could be included in the blog and we don't need to highlight it. Highlight the release date is useful for users, though.

Generally the picture looks good :)

**Reply by Anonymitaet** [reply]
The reason for highlighting release managers is that this might be an incentive for them (improve visibility).

**Comment by momo-jun** [comment]
Cross-post the [comment](https://github.com/apache/pulsar/issues/16938#issuecomment-1214783189) here since the Versions page will be deprecated, and the explanation for versioning change can be added to this new page.

**Comment by Anonymitaet** [comment]
We'll implement it as discussed above. I've created an issue to track the progress. https://github.com/apache/pulsar/issues/17514

**Comment by Anonymitaet** [comment]
Hi @eolivelli, 

We have an issue with 2.9.0 and 2.9.1 and find that you're the release manager for them.
Could you please give us a hand? 
Thank you!

🔹🔹🔹

### Context

We (@urfreespace, @SignorMercurio) are implementing this idea. 

The [Pulsar release table](https://pulsar.apache.org/release-notes/) is automatically generated based on the info of the [Pulsar GitHub release page](https://github.com/apache/pulsar/releases), where 
2.9.0 and 2.9.1 are missing. We're wondering why and figuring out how to deal with this situation.
<img width="1011" alt="image" src="https://user-images.githubusercontent.com/50226895/193180574-94ec2c19-ad76-46ed-b5e8-83bbea6a1d58.png">

### Questions

- 2.9.0: from the [2.9.0 release note](https://pulsar.staged.apache.org/release-notes/#290), we know it's a special release. Is this the reason for not showing 2.9.0 on the [Pulsar GitHub release page](https://github.com/apache/pulsar/releases)?
<img width="1276" alt="image" src="https://user-images.githubusercontent.com/50226895/193181694-dbe77820-a99c-4083-9665-e494378bca97.png">

- 2.9.1: from the [2.9.1 release note](https://pulsar.staged.apache.org/release-notes/#291), seems that this is a correct release. Then why is 2.9.1 missing on the [Pulsar GitHub release page](https://github.com/apache/pulsar/releases)? Should it be added?

**Comment by momo-jun** [comment]
@Anonymitaet FYI - The [release blog for 2.10.0](https://pulsar.apache.org/blog/2022/05/11/whats-new-in-pulsar-210) is in place but has not been shown on this new page yet.

**Reply by Anonymitaet** [reply]
> @Anonymitaet FYI - The [release blog for 2.10.0](https://pulsar.apache.org/blog/2022/05/11/whats-new-in-pulsar-210) is in place but has not been shown on this new page yet.

@SignorMercurio is this issue caused by website build failure?

**Reply by tisonkun** [reply]
Yes. The page is online now.
